mechanics/physicell/src/vtk_agents_serializer.cpp

```diff
@@ -0,0 +1,352 @@
+	orientation_array->SetNumberOfTuples(static_cast<vtkIdType>(agent_count));
+
+	for (auto& arr : cell_adhesion_affinity_arrays)
+	{
+		arr->SetNumberOfTuples(static_cast<vtkIdType>(agent_count));
+	}
+
+	for (auto& arr : chemotactic_sensitivity_arrays)
+	{
+		arr->SetNumberOfTuples(static_cast<vtkIdType>(agent_count));
+	}
+
+	for (index_t i = 0; i < agent_count; ++i)
+	{
+		radius_array->SetValue(static_cast<vtkIdType>(i), data.radius[i]);
+		cell_cell_adhesion_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.cell_cell_adhesion_strength[i]);
+		cell_BM_adhesion_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.cell_BM_adhesion_strength[i]);
+		cell_cell_repulsion_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.cell_cell_repulsion_strength[i]);
+		cell_BM_repulsion_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.cell_BM_repulsion_strength[i]);
+		relative_maximum_adhesion_distance_array->SetValue(static_cast<vtkIdType>(i),
+														   data.mechanics.relative_maximum_adhesion_distance[i]);
+		maximum_number_of_attachments_array->SetValue(
+			static_cast<vtkIdType>(i), static_cast<real_t>(data.mechanics.maximum_number_of_attachments[i]));
+		attachment_elastic_constant_array->SetValue(static_cast<vtkIdType>(i),
+													data.mechanics.attachment_elastic_constant[i]);
+		attachment_rate_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.attachment_rate[i]);
+		detachment_rate_array->SetValue(static_cast<vtkIdType>(i), data.mechanics.detachment_rate[i]);
+
+		is_motile_array->SetValue(static_cast<vtkIdType>(i), static_cast<real_t>(data.motility.is_motile[i]));
+		persistence_time_array->SetValue(static_cast<vtkIdType>(i), data.motility.persistence_time[i]);
+		migration_speed_array->SetValue(static_cast<vtkIdType>(i), data.motility.migration_speed[i]);
+		migration_bias_array->SetValue(static_cast<vtkIdType>(i), data.motility.migration_bias[i]);
+		restrict_to_2d_array->SetValue(static_cast<vtkIdType>(i), static_cast<real_t>(data.motility.restrict_to_2d[i]));
+		chemotaxis_index_array->SetValue(static_cast<vtkIdType>(i),
+										 static_cast<real_t>(data.motility.chemotaxis_index[i]));
+		chemotaxis_direction_array->SetValue(static_cast<vtkIdType>(i),
+											 static_cast<real_t>(data.motility.chemotaxis_direction[i]));
+		simple_pressure_array->SetValue(static_cast<vtkIdType>(i), data.state.simple_pressure[i]);
+		cell_definition_index_array->SetValue(static_cast<vtkIdType>(i),
+											  static_cast<real_t>(data.state.agent_type_index[i]));
+		is_movable_array->SetValue(static_cast<vtkIdType>(i), static_cast<real_t>(data.state.is_movable[i]));
+
+		std::array<double, 3> velocity = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> previous_velocity = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> migration_bias_dir = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> motility_vector = { 0.0, 0.0, 0.0 };
+		std::array<double, 3> orientation = { 0.0, 0.0, 0.0 };
```

Comment:
please, use std::array when the size is static. this results in needless 6 allocations